### ISSUE TITLE
Mac version

### ISSUE THREAD

**Issue by skumor-foreflight** [Frodo45127/rpfm#293]
Is it feasible to try to compile this for mac? I have tried to use the windows version with crossover/wine but that is based around exe files of the games so I am unable to make that work. If there is some other mac users maybe you can reach out to me and we can try to make it work together ?

**Comment by Frodo45127** [comment]
yes, it should be possible to compile it in mac (I managed to do it in a hackintosh vm a while ago), but there are some code changes needed in order to make it work with mac games:
- rpfm_lib\src\games\supported_games.rs: In each game there that has a mac version, it needs a "InstallData" section with the data of the mac's version of the game. Like these:
![imagen](https://github.com/user-attachments/assets/2e74fb4b-457f-44fc-bcf2-afd549ca79a4)
As for what each of those means: 
![imagen](https://github.com/user-attachments/assets/977ece9c-896b-49ac-8636-df7bce6aa25d)
- rpfm_lib\src\games\mod.rs: The install_type function has to be changed in order to dectect the new mac-specific InstallData sections.
- The build script may need changes in the mac-specific function (not sure).
- The release/test scripts may need to be expanded to cover mac too.
- All files have to be checked in case there is a rogue "if cfg!(target="windows")" that needs a macos else.
- And in my case, I would need a vm that actually works without horrible performance to test the builds, as I don't have a mac.

With just the first two changes any RPFM build through crossover should detect the mac games correctly, though you may need to specify their path manually in the settings. The rest of the stuff is needed for me to provide official mac builds.

**Comment by skumor-foreflight** [comment]
@Frodo45127 thank you, I will look into it

**Comment by Frodo45127** [comment]
> Hi [@Frodo45127](https://github.com/Frodo45127)! Are there plans to provide Mac packages or a complete compilation instruction?

None, at least by me, because I don't have a mac, and last time I tried setting up a mac VM it had abysmal performance. And about compilation instruction, it's more or less the same as in linux. Just need to figure out how to install the dependencies.
